### PR TITLE
Feature/profile page design

### LINKED ISSUE #607: Dashboard: Profile Main section UI
**Describe the solution you'd like**
Implementation of the attached UI of the Profile Section in the Profile screen based on code guidelines.

**Additional context**
Refer [Figma](https://www.figma.com/file/BrP5RowC7va50HyETZQUDA/Ignitus-Design?node-id=121%3A8) for User-Interface and other styles-related issue.

- You can view all the CSS, colors, typography, shadow effects, required to implement the design.
- Export the png icons from the design.

profile information

Dropdown UI for More option

Capture

Please make the button, cards, etc as a shared component that could be used all over the app to make the app consistent. Text is already defined in the code which should be used in this component.

### PR DESCRIPTION
**Pull request by ashishnagpal2498**
fixes #607 
Profile page Design - preview -
1) Can be viewed on profileList Route for now
2) Icons - bookmark , chat and resume to be added and changed
3) Only list item will be returned as Component - UL is created just to show for now
4) Dummy Image added

### CODE REVIEW

File: src/ignitus-Shared/Components/ProfileCard/Styles/index.scss [hunk 1]
```diff
@@ -0,0 +1,168 @@
+$base-color: #000066;
+$white-bg: #fff;
+
+.list-item {
+  padding: 2em 1em;
+  display: flex;
+  flex-wrap: wrap;
+  box-shadow: 0 2px 5px 5px rgba(0,0,0,0.1);
```
Comment (Paarmita) [inline src/ignitus-Shared/Components/ProfileCard/Styles/index.scss]: Put box-shadow as this `box-shadow: 0 1px 8px 1px rgba(0, 0, 0, 0.1); `
Make this card CSS in a shared component.

File: src/ignitus-Shared/Components/ProfileCard/Styles/index.scss [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+ }
+}
+
+.card-links{
+  text-align: left;
+  margin: 1em 0;
+  i {
+    color: $base-color;
+    font-size: 2em;
+    margin: 0 2em 0 0;
```
Comment (Paarmita) [inline src/ignitus-Shared/Components/ProfileCard/Styles/index.scss]: The margin between social icons looks large, so can you please make it 1 rem. `margin: 0 1em 0 0;`

File: src/ignitus-Shared/Components/ProfileCard/Styles/index.scss [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+}
+
+// Dropdown Button
+
+.btn-outer {
+  position: relative;
+}
+
+.custom-dropdown-btn {
+  color: $base-color;
```
Comment (Paarmita) [inline src/ignitus-Shared/Components/ProfileCard/Styles/index.scss]: `More..` should be bold and #939393 hex code.

File: src/ignitus-Shared/Components/ProfileCard/Styles/index.scss (hunk 1 above)
Comment (ashishnagpal2498) [reply src/ignitus-Shared/Components/ProfileCard/Styles/index.scss]: I have put this card in shared component . By "Make this card CSS in a shared component " what exactly you mean?

**Comment by ashishnagpal2498** [comment]
@gittysachin Yes , Icons have to be added , which I mentioned in the PR description. I didn't get the other thing you are trying to explain. "Round the top button in top left corner and even when hover over first button after hovering the more tab. Same applies for the bottom one as shown in the figma design."

File: src/ignitus-Shared/Components/ProfileCard/DropDownButton.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import {withErrorBoundary} from '../errorBoundary';
+// import loader from '../../ignitus-Assets/Images/loader2.gif';
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/ProfileCard/DropDownButton.js]: Remove comments here.

File: src/ignitus-Shared/Components/ProfileCard/Styles/index.scss [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    }
+  }
+}
+
+.custom-dropdown-menu {
+  z-index: 3;
+  position: absolute;
+  top: 30px;
+  left: 0;
+  border-radius: 10px 0 10px 0;
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/ProfileCard/Styles/index.scss]: Can be written as `10px 0`.

File: src/ignitus-Shared/Components/ProfileCard/Styles/index.scss [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+@media screen and (max-width: "768px")
+{
+  .list-item {
+    justify-content: center;
+  }
+
+  .profile-details,
+  .profile-contact {
+    flex-basis: 100%;
+    max-width: 100%;
```
[the remark this answers is not in the document]
Comment (divyanshu-rawat) [reply src/ignitus-Shared/Components/ProfileCard/Styles/index.scss]: Resolve all `hound` suggestions.

File: src/ignitus-Shared/Components/ProfileCard/index.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  <div className="card-links">
+    <i className="fa fa-linkedin-square" />
+    <i className="fa fa-facebook-f" />
+    <i className="fa fa-github" />
+    <i className="fa fa-google-plus" />
+  </div>
+);
+
+export const ProfileCard = () => (
+  <ul style={ { margin: '150px 60px' } }>
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/ProfileCard/index.js]: Remove `inline-styling`.

File: src/ignitus-Shared/Components/ProfileCard/index.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    <li className="list-item">
+      <div className="profile-details">
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/ProfileCard/index.js]: use `camelCase` for all classNames.

**Comment by divyanshu-rawat** [comment]
Hint that maybe be useful to visualise `columns` and `rows`.

76422570-b4a9c500-63a5-11ea-8c31-5864e56a5710

File: src/ignitus-Shared/Components/ProfileCard/Styles/index.scss [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+  padding-left: 150px;
+  padding-top: 20px;
+  flex-basis: 50%;
+  max-width: 50%;
+  position: relative;
+}
+
+.avatar-div {
+  left: 0;
+  position: absolute;
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/ProfileCard/Styles/index.scss]: Please don't use `position: absolute;  left: 0; top: 10px;` I don't understand their use-case they are generally helpful in layering content over each other like if we have to use `z-index` etc.

What is the use-case of this in this design?

File: src/ignitus-Shared/Components/ProfileCard/Styles/index.scss [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+  margin-right: 20px;
+  width: 150px;
+}
+
+.profile-avatar {
+  left: 50%;
+  position: absolute;
+  top: 50%;
+  transform: translate(-50%, -50%);
+  width: 120px;
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/ProfileCard/Styles/index.scss]: Same here Please don't use `position: absolute;  left: 0; top: 10px;` I don't understand their use-case they are generally helpful in layering content over each other like if we have to use `z-index` etc.

**Comment by divyanshu-rawat** [comment]
cc @gittysachin , @Paarmita , @debck

File: src/ignitus-Shared/Components/ProfileCard/Styles/index.scss (hunk 9 above)
Comment (ashishnagpal2498) [reply src/ignitus-Shared/Components/ProfileCard/Styles/index.scss]: The use of position is to fix the size of image. Making it flex will shrink it according to size of width. That is why position is used

File: src/ignitus-Shared/Components/ProfileCard/index.js (hunk 8 above)
Comment (ashishnagpal2498) [reply src/ignitus-Shared/Components/ProfileCard/index.js]: camelCase gives hound warnings.

**Comment by ashishnagpal2498** [comment]
Code changed to flex. Same icon issue , for now I added "+" icons. CamelCase of classNames gives hound warnings.

**Comment by ashishnagpal2498** [comment]
I have gone through the changes you did. Amazing , your code is more refined. My purpose of dividing into flex was to change view on smaller widths. I think if we have to add media query later so why no now ? 
image

File: src/ignitus-Shared/Components/ProfileCard/Styles/index.scss [hunk 11]
```diff
@@ -0,0 +1,206 @@
...
+.cardOuter {
+  display: flex;
+  flex-wrap: wrap;
+  flex-direction: row;
+  padding: 20px;
+  margin: 200px 30px;
+  justify-content: center;
+  background-color: $white-bg;
+  color: $base-color;
+  position: relative;
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/ProfileCard/Styles/index.scss]: what `position:relative `is doing here, in `flex` based layout using position doesn't make much sense.?

File: src/ignitus-Shared/Components/ProfileCard/Styles/index.scss [hunk 12]
```diff
@@ -0,0 +1,206 @@
...
+  border-radius: 15px;
+}
+
+.cardMain {
+  flex-basis: 70%;
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/ProfileCard/Styles/index.scss]: Instead you can use `flex: 5 or flex:6` to distribute the content over larger length.

File: src/ignitus-Shared/Components/ProfileCard/Styles/index.scss [hunk 13]
```diff
@@ -0,0 +1,206 @@
...
+  flex-basis: 100%;
+}
+
+.contactInfo {
+  flex-basis: 100%;
+}
+
+.textIcon {
+  padding-left: 20px;
+  position: relative;
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/ProfileCard/Styles/index.scss]: again `position:relative`, I can design this layout without using position anywhere in the css.

File: src/ignitus-Shared/Components/ProfileCard/Styles/index.scss [hunk 14]
```diff
@@ -0,0 +1,206 @@
...
+  vertical-align: bottom;
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/ProfileCard/Styles/index.scss]: In flex based layout, `justify-content` & `align-items` take care of horizontal and vertical alignment. read - https://stackoverflow.com/questions/19026884/flexbox-center-horizontally-and-vertically

File: src/ignitus-Shared/Components/ProfileCard/Styles/index.scss [hunk 15]
```diff
@@ -0,0 +1,206 @@
...
+  
+  span {
+    display: inline-block;
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/ProfileCard/Styles/index.scss]: please try to use flex based layout.

File: src/ignitus-Shared/Components/ProfileCard/Styles/index.scss [hunk 16]
```diff
@@ -0,0 +1,206 @@
...
+    width: 20px;
+    left: 0;
+    position: absolute;
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/ProfileCard/Styles/index.scss]: `position:absolute` should be eliminated if you use `flex-based `layout or `css grid `up-to you. ;)

File: src/ignitus-Shared/Components/ProfileCard/Styles/index.scss [hunk 17]
```diff
@@ -0,0 +1,206 @@
...
+    top: 0;
+    height: 100%;
+    
+    i {
+      font-size: 1.2em;
+      font-weight: 500;
+      left: 0;
+      position: absolute;
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/ProfileCard/Styles/index.scss]: same here.

File: src/ignitus-Shared/Components/ProfileCard/Styles/index.scss [hunk 18]
```diff
@@ -0,0 +1,206 @@
...
+      top: 0;
+      }
+  }
+}
+
+.card-links {
+  text-align: left;
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/ProfileCard/Styles/index.scss]: relying on grid / flex should eliminate this css property.

File: src/ignitus-Shared/Components/ProfileCard/Styles/index.scss [hunk 19]
```diff
@@ -0,0 +1,206 @@
...
+  margin: 1em 0;
+  i {
+    color: $base-color;
+    font-size: 2em;
+    margin-right: 1em;
+  }
+}
+
+.trackPerson {
+  position: absolute;
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/ProfileCard/Styles/index.scss]: same here. please read - https://teamtreehouse.com/community/when-do-i-use-absolute-positioning-and-when-do-i-use-relative-positioning

File: src/ignitus-Shared/Components/ProfileCard/Styles/index.scss [hunk 20]
```diff
@@ -0,0 +1,206 @@
...
+  right: 20px;
+  top: 0;
+  padding: 5px;
+  
+  p {
+    padding-right: 40px;
+    vertical-align: middle;
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/ProfileCard/Styles/index.scss]: https://stackoverflow.com/questions/19026884/flexbox-center-horizontally-and-vertically

File: src/ignitus-Shared/Components/ProfileCard/Styles/index.scss [hunk 21]
```diff
@@ -0,0 +1,206 @@
...
+  &:hover {
+
+    .custom-dropdown-menu {
+      display: block;
+    }
+  }
+}
+
+.custom-dropdown-menu {
+  z-index: 3;
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/ProfileCard/Styles/index.scss]: Here `z index` makes sense and `position:absolute` as well. good job 👍 but please understand that you don't have to use `position:relative` everywhere.

File: src/ignitus-Shared/Components/ProfileCard/index.js [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+    <i className="fa fa-google-plus" />
+  </div>
+);
+
+export default function ProfileCard() {
+  return (
+    <div className="cardOuter">
+      <div className="cardMain">
+        <div className="cardImage">
+          <img src={require('./dummy_image.png')} alt="profile-avatar" />
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/ProfileCard/index.js]: Please omit using require in rely on `ES6 modules`.

File: src/ignitus-Shared/Components/ProfileCard/index.js [hunk 23]
```diff
@@ -0,0 +1,59 @@
...
+        <div className="contactInfo">
+          <p className="textIcon">
+            <span>
+              <i className="fa fa-plus" />
+            </span>
+            Send them a message
+          </p>
+          <p className="textIcon">
+            <span>
+              <i className="fa fa-plus" />
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/ProfileCard/index.js]: I would suggest to use to follow this pattern to display icons, https://github.com/Ignitus/Ignitus-client/tree/develop/src/ignitus-Shared/ignitus-DesignSystem/ignitus-Assets/ignitus-Icons
Use f`ont awesome` / `material icons` to download the `svg`.

File: src/ignitus-Shared/Components/ProfileCard/index.js [hunk 24]
```diff
@@ -0,0 +1,59 @@
...
+      <div className="cardMain">
+        <div className="cardImage">
+          <img src={require('./dummy_image.png')} alt="profile-avatar" />
+        </div>
+        <div className="cardInfo">
+          <h2>Sophia Carter</h2>
+          <p>Literature Student</p>
+          <p className="textIcon">
+            <span>
+              <i className="fa fa-map-marker" />
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/ProfileCard/index.js]: I would suggest to use to follow this pattern to display icons, https://github.com/Ignitus/Ignitus-client/tree/develop/src/ignitus-Shared/ignitus-DesignSystem/ignitus-Assets/ignitus-Icons
Use `font awesome` / `material icons` to download the svg.

File: src/ignitus-Shared/Components/ProfileCard/index.js [hunk 25]
```diff
@@ -0,0 +1,59 @@
+/* eslint-disable jsx-a11y/accessible-emoji */
+import React from 'react';
+import './Styles/index.scss';
+import DropDownButton from './DropDownButton';
+
+export const SocialIcons = () => (
+  <div className="card-links">
+    <i className="fa fa-linkedin-square" />
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/Components/ProfileCard/index.js]: I would suggest to use to follow this pattern to display icons, https://github.com/Ignitus/Ignitus-client/tree/develop/src/ignitus-Shared/ignitus-DesignSystem/ignitus-Assets/ignitus-Icons
Use `font awesome` / `material icons` to download the svg.

**Comment by divyanshu-rawat** [comment]
@ashishnagpal2498 please don't create a PR directly to `master`, create a one to `develop`.  I am changing the base now,

**Comment by divyanshu-rawat** [comment]
> I have gone through the changes you did. Amazing , your code is more refined. My purpose of dividing into flex on width was to change view on smaller widths. I think if we have to add media query later so why no now ?
> 576px
> image
> 
> greater pixels
> image

@ashishnagpal2498 changing `flex-direction` on different screen sizes can do wonders themselves, instead of relying on` media-Queries`, and the way you were writing them it was not responsive at all.

**Comment by ashishnagpal2498** [comment]
Okay , will work on the changes.

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Assets/ignitus-Icons/icon.tsx [hunk 26]
```diff
@@ -0,0 +1,72 @@
...
+    case AppIcon.FileIcon:
+      return <FileIcon {...rest} />;
+
+    case AppIcon.MapMarkerIcon:
+      return <MapMarkerIcon {...rest} />;
+
+    case AppIcon.BookmarkIcon:
+      return <BookmarkIcon {...rest} />;
+
+    case AppIcon.ChatMessageIcon:
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/ignitus-DesignSystem/ignitus-Assets/ignitus-Icons/icon.tsx]: Adding Icons Looks Good 👍

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Organisms/ignitus-ProfileCard/Components/ProfileCard.jsx [hunk 27]
```diff
@@ -0,0 +1,67 @@
+/* eslint-disable jsx-a11y/accessible-emoji */
+import React from 'react';
+import * as S from '../styles';
+import { TextIconContainer } from '../../../shared';
+
+export default function ProfileCard() {
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/ignitus-DesignSystem/ignitus-Organisms/ignitus-ProfileCard/Components/ProfileCard.jsx]: can be re-written as export const profileCard = () => ( ..here goes the html ).

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Atoms/colors.ts [hunk 28]
```diff
@@ -21,3 +21,4 @@ export const Yellow: string = '#ffd747';
 // social platform colors
 export const TwitterIconColor: string = ' #1dcaff';
 export const LinkedinIconColor: string = '#2867B2';
+export const Grey3: string = '#939393';
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/ignitus-DesignSystem/ignitus-Atoms/colors.ts]: use `GreyText` here instead.

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Organisms/ignitus-ProfileCard/Components/ProfileCard.jsx [hunk 29]
```diff
@@ -0,0 +1,67 @@
...
+            <TextIconContainer>
+              <S.LocationIcon />
+              Location
+            </TextIconContainer>
+            <DropDownButton />
+          </S.DetailsContainer>
+        </S.ContentDiv>
+        <S.SocialDiv>
+          <S.SocialIcons>
+            <i className="fa fa-linkedin-square" />
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/ignitus-DesignSystem/ignitus-Organisms/ignitus-ProfileCard/Components/ProfileCard.jsx]: Please don't use fontAwesome use SVG Icons. 🔨

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Organisms/ignitus-ProfileCard/Components/ProfileCard.jsx [hunk 30]
```diff
@@ -0,0 +1,67 @@
...
+            <i className="fa fa-facebook-f" />
+            <i className="fa fa-github" />
+            <i className="fa fa-google-plus" />
+          </S.SocialIcons>
+          <S.SocialContact>
+            <TextIconContainer>
+              <S.SendMessageIcon />
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/ignitus-DesignSystem/ignitus-Organisms/ignitus-ProfileCard/Components/ProfileCard.jsx]: Use this approach - https://github.com/Ignitus/Ignitus-client/blob/master/src/ignitus-AboutPage/ignitus-CoreTeam/Components/team.tsx#L33

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Organisms/ignitus-ProfileCard/styles.ts [hunk 31]
```diff
@@ -0,0 +1,180 @@
+import styled from '@emotion/styled';
+import * as C from '../../ignitus-Atoms/colors';
+import { flexibleColDiv, flexibleRowDiv, TextIconContainer } from '../../shared';
+import ChatMessageIcon from '../../ignitus-Assets/ignitus-Icons/chatMessageIcon/chatMessageIcon';
+import FileIcon from '../../ignitus-Assets/ignitus-Icons/fileIcon/fileIcon';
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/ignitus-DesignSystem/ignitus-Organisms/ignitus-ProfileCard/styles.ts]: Don't import every Icon, use this approach defined here : https://github.com/Ignitus/Ignitus-client/blob/master/src/ignitus-AboutPage/ignitus-CoreTeam/Styles/index.ts#L100

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Organisms/ignitus-ProfileCard/styles.ts [hunk 32]
```diff
@@ -0,0 +1,180 @@
...
+    flex: 1 1 100%;
+  }
+`;
+
+export const ProfileBookmarkIcon = styled(BookmarkIcon)`
+  height: 2rem;
+  fill: ${C.IgnitusBlue};
+`;
+
+export const SendMessageIcon = styled(ChatMessageIcon)`
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/ignitus-DesignSystem/ignitus-Organisms/ignitus-ProfileCard/styles.ts]: Don't import every Icon, use this approach defined here : https://github.com/Ignitus/Ignitus-client/blob/master/src/ignitus-AboutPage/ignitus-CoreTeam/Styles/index.ts#L100

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Organisms/ignitus-ProfileCard/styles.ts [hunk 33]
```diff
@@ -0,0 +1,180 @@
...
+  height: 1.5rem;
+`;
+
+export const ResumeIcon = styled(FileIcon)`
+  height: 1.5rem;
+`;
+
+export const LocationIcon = styled(MapMarkerIcon)`
+  height: 1.2rem;
+  position: relative;
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/ignitus-DesignSystem/ignitus-Organisms/ignitus-ProfileCard/styles.ts]: I still don't why you always use position property at all everything can be designed without using it at all, it makes sense when you are planning to use `z-index.`

When removing `position:relative` distorts  the UI badly.

Screenshot 2020-03-25 at 01 35 48


You can align `icon` with the `text` in a very simple manner, by having a flex box defined as.

```
    display: flex;
    flex-direction: row;
```
```
  svg {
    left: 0;
    position: absolute;
    top: 0;
    font-size: 1em;
		fill: ${C.IgnitusBlue};
  }
```
along with removal of css that you have defined on `SVG`,     

Screenshot 2020-03-25 at 01 37 53

File: src/ignitus-Shared/ignitus-DesignSystem/shared.ts [hunk 34]
```diff
@@ -91,4 +91,27 @@ export const flexibleRowDiv = styled.div`
...
+  width: 100%;
+  text-align: left;
+  span {
+    font-weight: 400;
+    padding-bottom: 5px;
+    display: inline-block;
+    cursor: pointer;
+  }
+  i,
+  svg {
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/ignitus-DesignSystem/shared.ts]: Please remove this    
```
    left: 0;
    position: absolute;
    top: 0;
```

File: src/ignitus-Shared/ignitus-DesignSystem/shared.ts [hunk 35]
```diff
@@ -91,4 +91,27 @@ export const flexibleRowDiv = styled.div`
   justify-content: center;
 `;
 
+export const TextIconContainer = styled.div`
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/ignitus-DesignSystem/shared.ts]: For Inspiration please have a look here. http://www.ignitus.org/interface/sideNavigation, here icons are used along with the text, with no involvement of position property at all. Here is the code for the same https://github.com/Ignitus/Ignitus-client/blob/master/src/ignitus-Shared/ignitus-DesignSystem/ignitus-Organisms/ignitus-SideNavigation/ignitus-DashboardSideNavigation/Components/index.tsx#L39

**Comment by divyanshu-rawat** [comment]
@ashishnagpal2498 I have resolved the merge conflicts, and also reviewed your PR, please have a look 👍 , overall nice work ⚡️

File: src/ignitus-Shared/Components/ProfileCard/Styles/index.scss (hunk 9 above)
Comment (divyanshu-rawat) [reply src/ignitus-Shared/Components/ProfileCard/Styles/index.scss]: @ashishnagpal2498 which Images are you taking about, you have used the same for `svgs` as well.

**Comment by divyanshu-rawat** [comment]
@ashishnagpal2498 are you still working on this :)

**Comment by ashishnagpal2498** [comment]
@divyanshu-rawat yes , I am working. Will do the changes in few hours.

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Organisms/ignitus-ProfileCard/styles.ts (hunk 33 above)
Comment (ashishnagpal2498) [reply src/ignitus-Shared/ignitus-DesignSystem/ignitus-Organisms/ignitus-ProfileCard/styles.ts:119]: Icons are of different sizes , therefore to align the text with those icons in single row I use position. By positioning icons fix the icon with the applied padding to the text.

**Comment by ashishnagpal2498** [comment]
Hi @divyanshu-rawat , kindly review this PR.

**Comment by divyanshu-rawat** [comment]
Margin, padding doesn't seem to match with original implementation , icons text doesn't seem to align properly.

I have resolved the merge conflicts as well.

76422570-b4a9c500-63a5-11ea-8c31-5864e56a5710 (1)



Implemented UI.

Screenshot 2020-04-06 at 02 11 54

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Organisms/ignitus-ProfileCard/Components/ProfileCard.jsx [hunk 36]
```diff
@@ -0,0 +1,81 @@
...
+        <S.ImageContainer>
+          <img
+            src="https://storage.googleapis.com/ignitus_assets/ig-avatars/Team_avatar-eugene.png"
+            alt="ProfileLogo"
+          />
+        </S.ImageContainer>
+        <S.DetailsContainer>
+          <S.ProfileHeading>Sophia Carter</S.ProfileHeading>
+          <S.ProfileSubHeading>Literature Student</S.ProfileSubHeading>
+          <S.TextIconContainer>
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/ignitus-DesignSystem/ignitus-Organisms/ignitus-ProfileCard/Components/ProfileCard.jsx:27]: Create a component that will receive `content & icon & direction` as property.

Then reuse it wherever you have icons along with text.
In cases where you have icon before text direction property will come handy if you pass direction as reverse then  use flex-direction will become row-reverse.

To see how you can manipulate css by passing props in emotion refer here - https://github.com/Ignitus/Ignitus-client/blob/develop/src/ignitus-UserInterfaceBook/InterfaceSideNavigation/style.ts

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Organisms/ignitus-ProfileCard/styles.ts (hunk 33 above)
Comment (divyanshu-rawat) [reply src/ignitus-Shared/ignitus-DesignSystem/ignitus-Organisms/ignitus-ProfileCard/styles.ts:119]: but don't you know you can change the size of icons? apart from that form where did you learnt to prefer position css property instead changing icon sizes?

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Organisms/ignitus-ProfileCard/styles.ts [hunk 37]
```diff
@@ -0,0 +1,188 @@
...
+  fill: ${C.IgnitusBlue};
+`;
+
+export const SocialLinks = styled(Link)`
+  margin-right: 20px;
+`;
+
+export const LocationIcon = styled(Icon)`
+  height: 1.2rem;
+  position: relative;
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/ignitus-DesignSystem/ignitus-Organisms/ignitus-ProfileCard/styles.ts:119]: not needed, there are lot of examples where icons are aligned with text without using any property like `position`.  for instance see buttons with icon here  - https://deploy-preview-710--peaceful-nightingale-55a1b7.netlify.com/interface/buttons

File: src/ignitus-Shared/ignitus-DesignSystem/ignitus-Organisms/ignitus-ProfileCard/styles.ts [hunk 38]
```diff
@@ -0,0 +1,188 @@
...
+
+  &:hover {
+    ul {
+      display: block;
+    }
+  }
+`;
+
+export const DropDownMenu = styled.ul`
+  z-index: 50;
```
Comment (divyanshu-rawat) [inline src/ignitus-Shared/ignitus-DesignSystem/ignitus-Organisms/ignitus-ProfileCard/styles.ts:153]: do you really think you need value as 50 here, value 2 will do the job, in worst case 3.

**Comment by divyanshu-rawat** [comment]
@ashishnagpal2498 your work has been migrated to #769.

**Comment by ashishnagpal2498** [comment]
Okay, @divyanshu-rawat . Kindly add GSSOC label to that PR. Thank you

**Comment by divyanshu-rawat** [comment]
@ashishnagpal2498 I have added the label, but I would like to give you one suggestion don't run after the label even after lot of discussion on this PR, in the end I had to improve lot of things, please go through your commits and better improve..

**Comment by ashishnagpal2498** [comment]
@divyanshu-rawat Ya , I went through the changes. I will take care of this and improve next time. Thank you for your guidance.

**Pull request closed without merging**